src/playlist.js

```diff
@@ -5,7 +5,130 @@
   'use strict';
 
   var DEFAULT_TARGET_DURATION = 10;
-  var duration, seekable, segmentsDuration;
+  var accumulateDuration, ascendingNumeric, duration, intervalDuration, rangeDuration, seekable;
+
+  // Array.sort comparator to sort numbers in ascending order
+  ascendingNumeric = function(left, right) {
+    return left - right;
+  };
+
+  /**
+   * Returns the media duration for the segments between a start and
+   * exclusive end index. The start and end parameters are interpreted
+   * as indices into the currently available segments. This method
+   * does not calculate durations for segments that have expired.
+   * @param playlist {object} a media playlist object
+   * @param start {number} an inclusive lower boundary for the
+   * segments to examine.
+   * @param end {number} an exclusive upper boundary for the segments
+   * to examine.
+   * @param includeTrailingTime {boolean} if false, the interval between
+   * the final segment and the subsequent segment will not be included
+   * in the result
+   * @return {number} the duration between the start index and end
+   * index in seconds.
+   */
+  accumulateDuration = function(playlist, start, end, includeTrailingTime) {
```

Comment:
Basically, yes. It tries to find the longest span of segments that have PTS info available at the start and end. It does this by:

1. Splitting the interval into ranges that do not have internal discontinuities
1. For each of these ranges, finding the earliest and latest segment with PTS info
 1. Adding up the differences between the earliest PTS and the latest PTS in each range
 1. Adding up the M3U8 based durations for any segments in the range that aren't in the interval above